homeassistant/components/netatmo/sensor.py

```diff
@@ -145,6 +161,55 @@ def get_entities():
 
     async_add_entities(await hass.async_add_executor_job(get_entities), True)
 
+    @callback
+    def add_public_entities():
+        """Retrieve Netatmo public weather entities."""
+        entities = []
+        for area in entry.options.get(CONF_WEATHER_AREAS).values():
+            data = NetatmoPublicData(
+                auth,
+                lat_ne=area[CONF_LAT_NE],
+                lon_ne=area[CONF_LON_NE],
+                lat_sw=area[CONF_LAT_SW],
+                lon_sw=area[CONF_LON_SW],
+            )
+            for sensor_type in SUPPORTED_PUBLIC_SENSOR_TYPES:
+                entities.append(
+                    NetatmoPublicSensor(
+                        area[CONF_AREA_NAME],
+                        data,
+                        sensor_type,
+                        area[CONF_PUBLIC_MODE],
+                        area[CONF_SHOW_ON_MAP],
```

Comment:
I would just put in area and resolve it in the init of the sensor
```suggestion
                    NetatmoPublicSensor(area, data, sensor_type)
```